receiver/otlpreceiver/otlp.go

```diff
@@ -123,6 +124,12 @@ func (r *otlpReceiver) startProtocolServers(host component.Host) error {
 			otlpgrpc.RegisterLogsServer(r.serverGRPC, r.logReceiver)
 		}
 
+		if r.cfg.GRPC.Reflection != nil {
+			if r.cfg.GRPC.Reflection.Enabled {
+				reflection.Register(r.serverGRPC)
```

Comment:
The `ConfigGRPCSettings` struct is used by components other than the otlpreceiver. With this implementation, other components will have a `reflection.enabled` option available, but it won't do anything unless implemented. 

Unfortunately, gRPC does not expose this as a `ServerOption`, so the solution may have to be to introduce a `ToServer` method as it happens with the HTTPServer settings.

@jpkrohling (picking you since you were requested a review) what do you think is the best approach here?

Reply:
Agree, I think the cleanest would indeed have a ToServer.